Data/Vector.hs

```diff
@@ -1746,6 +1749,22 @@ scanr1' :: (a -> a -> a) -> Vector a -> Vector a
 {-# INLINE scanr1' #-}
 scanr1' = G.scanr1'
 
+-- Comparisons
+-- ------------------------
+
+-- | /O(n)/ Check if two vectors are equal using supplied equality
+-- predicate.
+eqBy :: (a -> a -> Bool) -> Vector a -> Vector a -> Bool
+{-# INLINE eqBy #-}
+eqBy = G.eqBy
+
+-- | /O(n)/ Compare two vectors using supplied comparison function for
+-- vector elements. Comparison works same as for lists.
+--
+-- > cmpBy compare == compare
+cmpBy :: (a -> a -> Ordering) -> Vector a -> Vector a -> Ordering
```

Comment:
Shouldn't it be 
```haskell
cmpBy :: (a -> b -> Ordering) -> Vector a -> Vector b -> Ordering
```
?

Reply:
Good catch. That is the type of `G.cmpBy`, so it seems like it should apply to specialized versions as well.

Reply:
I specialized these to `a -> a -> Ordering` because almost always vectors of the same type are compared and if someone needs most generic case there's Data.Vector.Generic.cmpBy

Reply:
I do not see a good reason to restrict specialized versions in such way. This is both less powerful and less expected.

Reply:
I thought that `a -> b -> Ordering` is somewhat unexpected. Anyway I changed specializations to `a -> b -> Ordering`

Reply:
Could you please do the same for `eqBy`?

Reply:
Done. Thank you for noticing things.